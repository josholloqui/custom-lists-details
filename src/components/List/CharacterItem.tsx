import * as React from 'react';

interface Props {
  name: string,
  image: string
}

const CharacterItem: React.FunctionComponent<Props> = ({ name, image }) => (
  <figure>
    <img src={image} alt={name} />
    <figcaption>{name}</figcaption>
  </figure>
);

export default CharacterItem;
