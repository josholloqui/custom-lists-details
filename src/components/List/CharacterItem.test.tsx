import * as React from 'react';
import { render, cleanup } from '@testing-library/react'
import CharacterItem from './CharacterItem';
import { MemoryRouter } from 'react-router-dom';

describe('CharacterItem Component', () => {
  afterEach(() => cleanup());

  it('render a CharacterItem', () => {
    const character = {
      id: 1,
      name: 'josh',
      image: 'imageurl'
    };

    const { asFragment } = render(
      <MemoryRouter>
        <CharacterItem {...character} />
      </MemoryRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
