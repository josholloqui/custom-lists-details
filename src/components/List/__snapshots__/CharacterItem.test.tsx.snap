// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterItem Component render a CharacterItem 1`] = `
<DocumentFragment>
  <figure>
    <img
      alt="josh"
      src="imageurl"
    />
    <figcaption>
      josh
    </figcaption>
  </figure>
</DocumentFragment>
`;
